nikola/shortcodes.py

```diff
@@ -106,28 +100,84 @@ def _find_shortcodes(data):
         yield [name, args, start, end]
 
 
-class SCParser(HTMLParser):
-    """Parser for shortcode arguments."""
-
-    # Because shortcode attributes are HTML-like, we are abusing the HTML parser.
-    # TODO replace with self-contained parser
-    # FIXME should be able to take quoted positional arguments!
-
-    def parse_sc(self, data):
-        """Parse shortcode arguments into a tuple."""
-        self.name = None
-        self.attrs = {}
-        self.feed(data)
-        args = []
-        kwargs = {}
-        for a, b in self.attrs:
-            if b is None:
-                args.append(a)
-            else:
-                kwargs[a] = b
-        return self.name, (args, kwargs)
-
-    def handle_starttag(self, tag, attrs):
-        """Set start tag information on parser object."""
-        self.name = tag
-        self.attrs = attrs
+def parse_sc(data):
+    """Parse shortcode arguments into a tuple."""
+    elements = data.split(' ', 1)
+    name = elements[0]
+    if len(elements) == 1:
+        # No arguments
+        return name, ([], {})
+    args = []
+    kwargs = {}
+
+    # "Simple" argument parser.
+    # flag can be one of:
+    # 0 name
+    # 1 value                               +value
+    # 2 name inside quotes                  +quotes
+    # 3 value inside quotes
+    # 4 [unsupported]                       +backslash
+    # 5 value inside backslash
+    # 4 [unsupported]
+    # 7 value inside quotes and backslash
+    flag = 0
+    cname = ''
+    cvalue = ''
+    qc = ''
+    for char in elements[1]:
+        if flag & 0b100 and flag & 1:
+            # Backslash in value: escape next character, no matter what
+            cvalue += char
+            flag -= 0b100
+        elif flag & 0b100:
+            # Backslash in name: escape next character, no matter what
+            cname += char
+            flag -= 0b100
+        elif char == '=' and flag == 0:
+            # Equals sign inside unquoted name: switch to value
+            flag = 1
+        elif char == ' ' and flag == 0:
+            # Space inside unquoted name: save as positional argument
+            args.append(cname)
+            cname = cvalue = qc = ''
+        elif char == ' ' and flag == 1:
```

Comment:
A bit too much cruft for me. If you really want that, you can make everything binary.